lib/src/concatenate.dart

```diff
@@ -82,6 +82,8 @@ extension Concatenate<T> on Stream<T> {
   /// Emits [initial] before any values or errors from the this stream.
   ///
   /// If this stream is a broadcast stream the result will be as well.
+  /// If this stream is a broadcast stream events which are emitted before
+  /// [initial] can be emitted asynchronously will be missed.
   Stream<T> startWith(T initial) =>
       startWithStream(Future.value(initial).asStream());
```

Comment:
(BTW, you can use `Stream<T>.value(initial)` instead of `Future<T>.value(initial).asStream()`. No promises that it has the precise same timing, but it *probably* does.)

Reply:
Done.

Reply:
It does end up changing timing in a way the tests are sensitive too - it looks like it makes it miss even more elements. I'd guess we used to deliver in a microtask, and now deliver in an event. I'll keep this implementation as is for now.